at4793-rk3994/project.md

```diff
@@ -0,0 +1,63 @@
+---
+title: "Kubernetes monitoring tools and techniques: A detailed analysis"
+author:
+  - Aakriti Talwar<at4793@nyu.edu>
+  - Ramneek Kaur <rk3994@nyu.edu>
+---
+
+> In what follows instructions (including this one) are in blockquotes, while
+> text not in blockquotes is a sample. You should of course replace the title
+> author metadata that shows up above with appropriate values.
+>
+> You should start your proposal by stating a problem statement or an overall
+> goal  as below.
+
+For our Big Data and Machine Learning final project we plan to present a deep dive analysis and comparison between top monitoring and reporting open source solutions available for Kubernetes clusters.
+
+# Motivation
+> Next you should talk about why, citing sources for your beliefs.
+
+The DevOps conceptual practice forces us to break significant problems into smaller problems. Microservices fit perfectly here, where small services build-up a component, and these components make up an application. The microservice architecture utilizes small teams to develop functional components one by one. By using a microservice architecture, it’s easy to make it faster, scale up and down without impacting the whole system. It’s also better for fault tolerance and is platform and language agnostic. Everything has pros and cons and for microservice architecture, it’s hard to maintain testing and monitoring.
+
+In microservice architecture, we deploy microservices as a container and for orchestration of containers, we rely heavily on tools like Kubernetes or Docker Swarm. Once deployed, microservice architecture has thousands of services talking to each other through networking, which can make it very challenging to monitor. We have to monitor independent services more than service to service communication. Kubernetes is widely growing in the market and it has a large rapidly growing ecosystem. Over the time many organizations are investing their resources in Kubernetes. Since it is open source, there are numerous plugins and tools available for analysis and reporting of the clusters. There are no definite comparisons between all of these tools and plugins available and neither do these organisations have time and resources to explore each of them individually. 
+
+Resources Used:
+https://logz.io/blog/kubernetes-monitoring/
+
+# Proposed Solution
+> Next you should talk about how you plan to address the problem. 
+> This should include information about what infrastructure (e.g., CloudLab
+> resources) you plan to use, how you plan to make progress, and how you plan to
+> evaluate your view.
+
+* Firstly, we will set up a working 3 node kubernetes cluster.
```

Comment:
Running what applications?

Reply:
We are planning on implementing few of the highly scalable applications given in this website: https://www.fromdev.com/2012/10/Open-Source-Projects-To-Build-Highly-Scalable-Web-Apps.html

Reply:
Those are frameworks, not applications. What workload?

Reply:
We are actually confused on which application to host.
Our requirement is just that we would be able to apply varying loads on the app to check the behaviour of the monitoring tools.( We preferrably plan to implement an application whose helm is readily available on the internet).
Could you please suggest us something here?

Reply:
I am a bit worried that this entire proposal is set up to carefully avoid writing any code, so at least for this bit I would suggest building yourself an interesting web service and then using https://github.com/wg/wrk or https://httpd.apache.org/docs/2.4/programs/ab.html to generate load. In my experience the utility of these tools varies depending on the communication pattern and application used, so you would probably want to write several.